ugs-core/src/com/willwinder/universalgcodesender/connection/ConnectionFactory.java

```diff
@@ -18,12 +18,44 @@ This file is part of Universal Gcode Sender (UGS).
  */
 package com.willwinder.universalgcodesender.connection;
 
+import com.willwinder.universalgcodesender.utils.Settings;
+import com.willwinder.universalgcodesender.utils.SettingsFactory;
+
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 /**
+ * A factory for creating a serial connection object using the settings
  *
  * @author wwinder
  */
 public class ConnectionFactory {
-    static public Connection getConnectionFor(String address, int baud) {
-        return new JSSCConnection();
+
+    private static final Logger logger = Logger.getLogger(ConnectionFactory.class.getSimpleName());
+
+    static public Connection getConnection() {
+        Settings settings = SettingsFactory.loadSettings();
```

Comment:
I couldn't find a better way to get the settings, it didn't feel right to pass it as a parameter this far down the chain. If you have any suggestions I will happily change it!